### PR TITLE
Improved the code of package org.jboss.forge.roaster

### PR DESCRIPTION
**Pull request by kaijMueller**
Hi,
I added a lot of javadoc comments to the sources in org.jboss.forge.roaster. In addition, I removed unused methods and improved the code quality if it was obvious (e.g. use try with resources).

Best regards,
Kai

### CODE REVIEW

File: api/src/main/java/org/jboss/forge/roaster/Roaster.java [hunk 1]
```diff
@@ -194,26 +262,36 @@ private Roaster()
...
          }
-         else if (unit != null)
-         {
-            throw new ParserException("Source does not represent a [" + type.getSimpleName() + "], instead was ["
-                     + unit.getGoverningType().getClass().getSimpleName() + "] - Cannot convert.");
-         }
+         throw new ParserException("Source does not represent a [" + type.getSimpleName() + "], instead was ["
+                  + unit.getGoverningType().getClass().getSimpleName() + "] - Cannot convert.");
       }
-      throw new ParserException("Cannot find JavaParser capable of parsing the requested data");
```
Comment (gastaldi) [inline api/src/main/java/org/jboss/forge/roaster/Roaster.java]: This is still required for cases when roaster-jdt is not in classpath

File: api/src/main/java/org/jboss/forge/roaster/Roaster.java [hunk 2]
```diff
@@ -93,36 +98,59 @@ private Roaster()
             return result;
          }
       }
-      throw new ParserException("Cannot find JavaParserProvider capable of producing JavaSource of type "
```
Comment (gastaldi) [inline api/src/main/java/org/jboss/forge/roaster/Roaster.java]: This is still required for cases when roaster-jdt is not in classpath

File: api/src/main/java/org/jboss/forge/roaster/Problem.java [hunk 3]
```diff
@@ -106,7 +112,6 @@ else if (!message.equals(other.message))
    @Override
    public String toString()
    {
-      return "Problem [message=" + message + ", sourceStart=" + sourceStart + ", sourceEnd=" + sourceEnd
-               + ", sourceLineNumber=" + sourceLineNumber + "]";
+      return "Problem: " + message;
```
Comment (gastaldi) [inline api/src/main/java/org/jboss/forge/roaster/Problem.java]: Maybe it is interesting to show the other attributes too?

File: api/src/main/java/org/jboss/forge/roaster/Roaster.java [hunk 4]
```diff
@@ -223,12 +301,18 @@ public static JavaUnit parseUnit(final String data)
          if (unit != null)
             return unit;
       }
-      throw new ParserException("Cannot find JavaParser capable of parsing the requested data");
```
Comment (gastaldi) [inline api/src/main/java/org/jboss/forge/roaster/Roaster.java]: This is still required for cases when roaster-jdt is not in classpath

File: api/src/main/java/org/jboss/forge/roaster/Roaster.java (hunk 1 above)
Comment (kaijMueller) [reply api/src/main/java/org/jboss/forge/roaster/Roaster.java]: Valid point. Will change this

File: api/src/main/java/org/jboss/forge/roaster/Roaster.java (hunk 2 above)
Comment (kaijMueller) [reply api/src/main/java/org/jboss/forge/roaster/Roaster.java]: Valid point. Will change this

File: api/src/main/java/org/jboss/forge/roaster/Problem.java (hunk 3 above)
Comment (kaijMueller) [reply api/src/main/java/org/jboss/forge/roaster/Problem.java]: Valid point. Will change this

File: api/src/main/java/org/jboss/forge/roaster/Roaster.java (hunk 4 above)
Comment (kaijMueller) [reply api/src/main/java/org/jboss/forge/roaster/Roaster.java]: Valid point. Will change this

File: api/src/main/java/org/jboss/forge/roaster/Roaster.java [hunk 5]
```diff
@@ -93,36 +99,58 @@ private Roaster()
             return result;
          }
       }
-      throw new ParserException("Cannot find JavaParserProvider capable of producing JavaSource of type "
-               + type.getSimpleName(), new IllegalArgumentException(type.getName()));
+      throw new ParserException("Cannot find JavaParser capable of parsing the requested data");
```
Comment (gastaldi) [inline api/src/main/java/org/jboss/forge/roaster/Roaster.java]: Why change the exception message here?

Comment (kaijMueller) [reply api/src/main/java/org/jboss/forge/roaster/Roaster.java]: Good question. Happened while coping code :-)

**Pull request merged**